lua/impairative/operations.lua

```diff
@@ -51,25 +52,40 @@ local function process_desc(desc, i)
     end
 end
 
-local function maybe_with_better_n(args, callbacks)
+---@param args table operation options
+---@param default_better_n_args ImpairativeBetterNArgs default better-n arguments
+---@param callbacks { backward: fun(), forward: fun() } callbacks to be wrapped
+---@return { backward: fun(), forward: fun() }
+local function maybe_with_better_n(args, default_better_n_args, callbacks)
     if not args.better_n then
         return callbacks
     else
-        local wrapper = require('better-n').create {
+        local better_n_args = type(args.better_n) == 'boolean'
+            and default_better_n_args
+            or vim.tbl_extend('force', default_better_n_args, args.better_n)
```

Comment:
Wouldn't this be a problem if `args.better_n` is `nil`?

Also, this is just my personal preference, but I really dislike Lua's ternary operator hack. I'd prefer to use an explicit if:

```lua
local better_n_args
if type(args.better_n) == 'boolean' then
    better_n_args = default_better_n_args
else
    better_n_args = vim.tbl_extend('force', default_better_n_args, args.better_n)
end
```

Reply:
It can't be `nil` since the enclosing if-else checked that already. But good that you mentioned it, I found control-flow to be non-obvious so I streamlined it.

Changed the style as you preferred.